service/grails-app/services/org/olf/rs/statemodel/events/EventISO18626IncomingAbstractService.groovy

```diff
@@ -145,9 +145,10 @@ public abstract class EventISO18626IncomingAbstractService extends AbstractEvent
                         processedSuccessfully = false;
                         errorType = ERROR_TYPE_UNABLE_TO_FIND_REQUEST;
                     } else {
-                        if ((request.supplyingInstitutionSymbol == null ||
+                        if (((request.supplyingInstitutionSymbol == null ||
                                 request.getSupplyingInstitutionSymbol().contains("null") ||
-                                    request.getSupplyingInstitutionSymbol() == ":") &&
+                                    request.getSupplyingInstitutionSymbol() == ":") ||
+                                EventMessageRequestIndService.isSlnpRequesterStateModel(request)) &&
                                         eventData.header?.supplyingAgencyId?.agencyIdType != null &&
```

Comment:
@JanisSaldabols This won't work. For SLNP we need to remove the supplying institution for Unfilled/Cancelled. So set it even if the new value is empty or null.

Reply:
Updated